Build is failing when importing statsmodels

From https://github.com/sgkit-dev/sgkit/actions/runs/15812962545/job/44566930125#step:6:59

```
_______________ ERROR collecting sgkit/tests/test_association.py _______________
ImportError while importing test module '/home/runner/work/sgkit/sgkit/sgkit/tests/test_association.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
sgkit/tests/test_association.py:41: in <module>
    import statsmodels.api as sm
/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/site-packages/statsmodels/api.py:76: in <module>
    from . import datasets, distributions, iolib, regression, robust, tools
/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/site-packages/statsmodels/distributions/__init__.py:7: in <module>
    from .discrete import (
/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/site-packages/statsmodels/distributions/discrete.py:5: in <module>
    from scipy._lib._util import _lazywhere
E   ImportError: cannot import name '_lazywhere' from 'scipy._lib._util' (/opt/hostedtoolcache/Python/3.11.12/x64/lib/python3.11/site-packages/scipy/_lib/_util.py)
```

Root cause is https://github.com/statsmodels/statsmodels/issues/9584

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -26,7 +26,7 @@ dependencies = [
   "xarray < 2025.03.1",
   "dask[array,dataframe] >= 2022.01.0, <= 2024.8.0",
   "distributed >= 2022.01.0, <= 2024.8.0",
-  "scipy",
+  "scipy < 1.16",
   "zarr >= 2.10.0, != 2.11.0, != 2.11.1, != 2.11.2, < 3",
   "numba",
   "typing-extensions",

--- a/requirements-numpy1.txt
+++ b/requirements-numpy1.txt
@@ -2,7 +2,7 @@ numpy < 2
 xarray < 2025.03.1
 dask[array,dataframe] >= 2023.01.0, <= 2024.8.0
 distributed >= 2023.01.0, <= 2024.8.0
-scipy
+scipy < 1.16
 typing-extensions
 numba
 zarr >= 2.10.0, != 2.11.0, != 2.11.1, != 2.11.2, < 3

--- a/requirements.txt
+++ b/requirements.txt
@@ -2,7 +2,7 @@ numpy < 2.2
 xarray < 2025.03.1
 dask[array,dataframe] >= 2023.01.0, <= 2024.8.0
 distributed >= 2023.01.0, <= 2024.8.0
-scipy
+scipy < 1.16
 typing-extensions
 numba
 zarr >= 2.10.0, != 2.11.0, != 2.11.1, != 2.11.2, < 3

